File: backend/src/main/kotlin/com/example/todo_app/Todo.kt
package com.example.todo_app

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime
import java.util.Date

@Table("todos")
data class Todo(
    @Id val id: Long? = null,
    val title: String,
    val description: String,
    val completed: Boolean = false,

    @CreatedDate
    @Column("created_at")
    val createdAt: LocalDateTime? = null,

    @LastModifiedDate
    @Column("updated_at")
    val updatedAt: LocalDateTime? = null
)

File: frontend/src/types/todo.ts
export interface ApiResponse<T> {
	success: boolean;
	data: T;
	error: string | null;
}

export interface TodoResponse {
	id: number;
	title: string;
	description: string;
	completed: boolean;
	createdAt: string;
	updatedAt: string;
}

export interface TodoListItemResponse {
	id: number;
	title: string;
	completed: boolean;
}

export interface PartialUpdateTodoRequest {
	title?: string;
	description?: string;
	completed?: boolean;
}
